packages/flutter_tools/test/general.shard/android/android_studio_validator_test.dart

```diff
@@ -17,59 +17,56 @@ import '../../src/context.dart';
     });
   });
 
-  group('AndroidStudioValidator', () {
-    MemoryFileSystem fs;
-    MockProcessManager mockProcessManager;
-    setUp(() {
-      fs = MemoryFileSystem();
-      mockProcessManager = MockProcessManager();
+  testUsingContext('AndroidStudioValidator gives doctor error on java crash', () async {
+    when(globals.processManager.canRun(any)).thenReturn(true);
+    when(globals.processManager.runSync(any)).thenAnswer((Invocation _) {
+      throw const ProcessException('java', <String>['--version']);
     });
+    const String installPath = '/opt/android-studio-with-cheese-5.0';
+    const String studioHome = '$home/.AndroidStudioWithCheese5.0';
+    const String homeFile = '$studioHome/system/.home';
+    globals.fs.directory(installPath).createSync(recursive: true);
+    globals.fs.file(homeFile).createSync(recursive: true);
+    globals.fs.file(homeFile).writeAsStringSync(installPath);
 
-    testUsingContext('gives doctor error on java crash', () async {
-      when(mockProcessManager.canRun(any)).thenReturn(true);
-      when(mockProcessManager.runSync(any)).thenAnswer((Invocation _) {
-        throw const ProcessException('java', <String>['--version']);
-      });
-      const String installPath = '/opt/android-studio-with-cheese-5.0';
-      const String studioHome = '$home/.AndroidStudioWithCheese5.0';
-      const String homeFile = '$studioHome/system/.home';
-      globals.fs.directory(installPath).createSync(recursive: true);
-      globals.fs.file(homeFile).createSync(recursive: true);
-      globals.fs.file(homeFile).writeAsStringSync(installPath);
-
-      // This checks that running the validator doesn't throw an unhandled
-      // exception and that the ProcessException makes it into the error
-      // message list.
-      for (final DoctorValidator validator in AndroidStudioValidator.allValidators) {
-        final ValidationResult result = await validator.validate();
-        expect(result.messages.where((ValidationMessage message) {
-          return message.isError && message.message.contains('ProcessException');
-        }).isNotEmpty, true);
-      }
-    }, overrides: <Type, Generator>{
-      FileSystem: () => fs,
-      ProcessManager: () => mockProcessManager,
-      Platform: () => linuxPlatform(),
-    });
+    // This checks that running the validator doesn't throw an unhandled
+    // exception and that the ProcessException makes it into the error
+    // message list.
+    for (final DoctorValidator validator in AndroidStudioValidator.allValidators) {
+      final ValidationResult result = await validator.validate();
+      expect(result.messages.where((ValidationMessage message) {
+        return message.isError && message.message.contains('ProcessException');
+      }).isNotEmpty, true);
+    }
+  }, overrides: <Type, Generator>{
+    FileSystem: () => fileSystem,
+    ProcessManager: () => MockProcessManager(),
+    Platform: () => linuxPlatform,
+    FileSystemUtils: () => FileSystemUtils(
```

Comment:
missing file system utils meant this was using the real android studio sometimes

Reply:
This is troubling...